src/js/helpers/mapController/miscLayerHelpers.ts

```diff
@@ -100,35 +100,83 @@ export async function extractWebmapLayerObjects(
       });
     } else {
       //TODO: This needs research, some layers have not only "id" but also "layerId" property. Those will differ, "id" will be "parent id for mapservice", and "layerId" will be its sublayer. Tricky part is that this happens with some layers on webmap in CMR, sublayers do not show on layer itself but the presense of layerId property indicates that it is indeed a sub
-      legendInfo = layer.layerId
-        ? legendInfo?.layers?.find((l: any) => l.layerId === layer.layerId)
-            .legend
-        : legendInfo;
-      const {
-        id,
-        title,
-        opacity,
-        visible,
-        definitionExpression,
-        url,
-        maxScale,
-        minScale
-      } = layer;
-      mapLayerObjects.push({
-        id,
-        title,
-        opacity,
-        visible,
-        definitionExpression,
-        group: 'webmap',
-        type: 'webmap',
-        origin: 'webmap',
-        url,
-        maxScale,
-        minScale,
-        sublayer: false,
-        legendInfo
-      });
+
+      if (legendInfo.error) {
+        const {
+          id,
+          title,
+          opacity,
+          visible,
+          definitionExpression,
+          url,
+          maxScale,
+          minScale
+        } = layer;
+        mapLayerObjects.push({
+          id,
+          title,
+          opacity,
+          visible,
+          definitionExpression,
+          group: 'webmap',
+          type: 'webmap',
+          origin: 'webmap',
+          url,
+          maxScale,
+          minScale,
+          sublayer: false,
+          portalItemID: layer.portalItem.id
+        });
+      } else {
+        legendInfo = layer.layerId
+          ? legendInfo.layers.find((l: any) => l.layerId === layer.layerId)
+              .legend
+          : legendInfo;
```

Comment:
can you pull in latest `clean-slate` and remove the changes from this file? I think I have a solution for this bug already in place!